Flyer test

JunAishima opened the pull request:
Creating objects for adding the flyer object used for standard collection (zebra + Galil vector + Pilatus) to LSDC. Pull requests and fixes while testing on the beamline.

RobertSchaffer1 commented:
This'll be a big one, I see.

JunAishima commented:
> This'll be a big one, I see.

l think this is a lot bigger than necessary - a lot of the code has already been reviewed, but we haven't merged into main yet. Let me see if I can try to clean things up a bit...

RobertSchaffer1 commented on nyxtools/_version.py:
```diff
@@ -190,27 +196,34 @@ def git_versions_from_keywords(keywords, tag_prefix, verbose):
         # sorting will prefer e.g. "2.0" over "2.0rc1"
         if ref.startswith(tag_prefix):
-            r = ref[len(tag_prefix):]
+            r = ref[len(tag_prefix) :]
```
As much as I hate this, I suppose it must be suggested by black/isort/flake?

mrakitin replied:
I did not see it in my latest commit https://github.com/NSLS-II-NYX/nyxtools/pull/10/commits/b96745f60da70ad23ef71d1a422a38a002555269, but probably was addressed by the earlier linting. And we usually use this `_version.py` file as is coming from https://github.com/NSLS-II/scientific-python-cookiecutter, so I would not worry about it too much. In some projects, we disable checking against `versioneer.py` and this file in the library.

RobertSchaffer1 commented:
> > This'll be a big one, I see.
> 
> l think this is a lot bigger than necessary - a lot of the code has already been reviewed, but we haven't merged into main yet. Let me see if I can try to clean things up a bit...

Ah, okay I'll wait for that then.

JunAishima commented:
UPDATED! I rebased on main after merging main with the last pull request ec6eb3d. This should contain only the changes done in the last 10 days or so and that have not been reviewed yet.

RobertSchaffer1 commented on nyxtools/flyer.py:
```diff
@@ -20,93 +23,183 @@ def __init__(self, vector, zebra, detector=None) -> None:
 
-        return NullStatus()
+        def zebra_callback(*args, **kwargs):
+            print(f"\n{ttime.ctime()}:\n\targs: {args}\n\tkwargs: {kwargs}\n")
```
We should replace these with logger.info or logger.debug statements, if we want to leave them in.

RobertSchaffer1 replied:
I can do that, if you guys agree.

mrakitin replied:
Yes, please go ahead, @RobertSchaffer1. I'll be happy to review your changes.

RobertSchaffer1 commented on nyxtools/flyer.py:
```diff
@@ -20,93 +23,183 @@ def __init__(self, vector, zebra, detector=None) -> None:
+                f"{self.detector.name}_image": {
+                    "source": f"{self.detector.name}_image",
+                    "dtype": "array",
+                    "shape": [-1, -1],  # TODO: read shapes from AD
```
I can do this, unless you're already on it @mrakitin

mrakitin replied:
Sounds great! Please have a look at https://github.com/bluesky/databroker/pull/596/files#diff-f7833108df15eaaac29030bc84ed7b307eb03b4eb91d959b4917d94ee1d13004R2097-R2101. That link has a few comments about what shape we expect (`(num_images, y, x)`).

RobertSchaffer1 commented on nyxtools/flyer.py:
```diff
@@ -20,93 +23,183 @@ def __init__(self, vector, zebra, detector=None) -> None:
+            now = ttime.time()
+            data = {
+                # f"{self.detector.name}_image": self._datum_ids["data"],
+                # "omega": self._datum_ids["omega"],
```
Do we want to get this back in?

mrakitin commented on nyxtools/vector.py:
```diff
@@ -232,28 +233,50 @@ def prepare_move(
+        run_status = SubscriptionStatus(self.state, start_callback, run=True)
+        print(f"{time.ctime()}: subscribed to {self.active.name}")
+
+        time.sleep(1.0)
```
Why did we need this 1s timeout here?

RobertSchaffer1 replied:
I can't think of a single good reason, which makes me think it's an artifact of the original code. I'll remove it.

mrakitin reviewed (approved):
I think it's in a good shape now. I posted a few items to address on Friday.

RobertSchaffer1 commented on nyxtools/vector.py:
```diff
@@ -232,28 +233,50 @@ def prepare_move(
+        def start_callback(value, old_value, **kwargs):
+            print(f"{time.ctime()}: {old_value} -> {value}")
+            if old_value == "Backup" and value == "Acquiring":
+                print(f"{time.ctime()}: Successfully changed {old_value} -> {value}")
```
While we're changing these to debugs we should remember to change this to ttime as well.

mrakitin replied:
Right, good point.

RobertSchaffer1 reviewed (approved):
Agreed w/ Maksim, most of this is fine. Could use some basic cleanup.

RobertSchaffer1 commented:
@JunAishima @mrakitin I made some minor changes, just the ones we discussed on this PR.

mrakitin reviewed (approved):
Looks good! Had a discussion with @RobertSchaffer1 about the log message format, and how to enable the `DEBUG` stdout logging on the client-side. This config may be useful:
https://github.com/NSLS-II/sirepo-bluesky/blob/0aed9b9c735c94728ce360345f09b6c4722bb583/sirepo_bluesky/sirepo_ophyd.py#L20-L24
